### PR TITLE
[core][docs] Document port/IP binding and slurm concerns

### LINKED ISSUE #10154: [core] [help] Running `ray start` on the same node in parallel would get port error
Hi,

I want to ask how is the port chosen and used for worker, within the range of [min_worker_port, max_worker_port]? Is it chosen randomly within this range? (Sorry that I don't go into the details of ray core and need some help on the implementation.)

When I call `ray start` for a raylet, it is starting one worker and one port is needed?

Does the following two errors mean the same thing?
```
# This error message is in 0.8.6; seems to be removed by 0.8.7?
(pid=1024090) F0817 16:07:46.085644 1024090 1024090 core_worker.cc:294]  Check failed: assigned_port != -1 Failed to allocate a port for the worker. Please specify a wider port range using the '--min-worker-port' and '--max-worker-port' arguments to 'ray start'.
(pid=1024090) *** Check failure stack trace: ***

# This is the root error of duplicate ports?
(pid=raylet) terminate called after throwing an instance of 'boost::wrapexcept<boost::system::system_error>'
(pid=raylet)   what():  bind: Address already in use
(pid=raylet) *** Aborted at 1597694865 (unix time) try "date -d @1597694865" if you are using GNU date ***
(pid=raylet) PC: @                0x0 (unknown)
(pid=raylet) *** SIGABRT (@0x3e8000f9fdf) received by PID 1023967 (TID 0x7fbbf1e81800) from PID 1023967; stack trace: ***
```

Thanks so much for your help in advance!

### PR DESCRIPTION
**Pull request by tupui**
## Why are these changes needed?

Using Ray on SLURM system is documented but missing some pitfalls about network. This PR adds some information about port binding and address binding (I will open a feature request with more and link it here later).

I did not put any real recommendation on this last point since `--address` did not work. I had cannot resolve issue after setting an internal IP although it's reachable.

## Related issue number

Closes #10154 and is a recurring theme on the forum.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/slurm.rst [hunk 1]
```diff
@@ -141,6 +147,40 @@ Finally, you can invoke your Python script:
...
+
+Considering 2 users, if they both schedule a SLURM job using Ray
+at the same time, they are both creating a head node. In the backend, Ray will
+assign some internal ports to a few services. The issue is that as soon as the
+first head node is created, it will bind some ports and prevent them to be
+used by another head node. To prevent any conflicts, users have to manually
+specify non overlapping ranges of ports. The following ports are to be
+adjusted::
+
+    --node-manager-port
```
Comment (pcmoritz) [inline doc/source/cluster/slurm.rst:173]: Can you give a concrete example here on how to set these parameters? (e.g. the full `ray start` commands to run or which settings to overwrite in the configuration) Basically the user should be able to just follow the instructions and make it work, without having to figure out a lot of stuff themselves :)

File: doc/source/cluster/slurm.rst [hunk 2]
```diff
@@ -141,6 +147,40 @@ Finally, you can invoke your Python script:
...
+    --object-manager-port
+    --port
+    --gcs-server-port
+    --min-worker-port
+    --max-worker-port
+
+As for the IP binding, on some cluster architecture the network interfaces
+do not allow to use external IPs between nodes. Instead, there are internal
+network interfaces (`eth0`, `eth1`, etc.). Currently, it's difficult to
+set an internal IP.
```
Comment (richardliaw) [inline doc/source/cluster/slurm.rst:183]: Can you add a link to the issue where we are tracking this?

Comment (tupui) [reply doc/source/cluster/slurm.rst]: Done

File: doc/source/cluster/slurm.rst (hunk 1 above)
Comment (tupui) [reply doc/source/cluster/slurm.rst:174]: Done

**Review by pcmoritz — APPROVED** [review]

**Pull request merged**